### PR TITLE
(PCP-244) Remove Makefile; add building instructions to README

### PR DESCRIPTION
**Pull request by parisiale**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,26 +11,106 @@ cpp-pcp-client implements PCP by layering it upon WebSocket; it uses
...
+
+Build and install
+-----------------
+
+* build & install [leatherman][leatherman]
+
+* build and install cpp-pcp-client
+
+  Thanks to the CMake, the project can be built out-of-source tree, which allows
+  for multiple independent builds.
```
Comment (mruzicka) [inline README.md]: Do you want to mention the ```DEV_LOG_RAW_MESSAGE``` option?

File: README.md [hunk 2]
```diff
@@ -13,26 +11,106 @@ cpp-pcp-client implements PCP by layering it upon WebSocket; it uses
...
+      make
+      sudo make install
 
-Tests can be run with:
+  example debug/test build:
 
-    make test
+      mkdir debug
+      cd debug
+      cmake -DCMAKE_BUILD_TYPE=Debug ..
```
Comment (mruzicka) [inline README.md]: perhaps add the ```-DDEV_LOG_RAW_MESSAGE=on``` here?

File: README.md (hunk 1 above)
Comment (parisiale) [inline README.md]: No, it's just for devs, as it's not safe to use it on Windows.
No publicity is better.

File: README.md (hunk 2 above)
Comment (parisiale) [inline README.md]: No, see above.

File: README.md (hunk 1 above)
Comment (mruzicka) [inline README.md]: OK

**Comment by mruzicka** [comment]
LGTM

**Pull request merged**